core/src/main/scala/dagr/core/execsystem/TaskManager.scala

```diff
@@ -47,12 +47,12 @@ object SystemResources {
   /** Creates a new SystemResources with the cores provided and partitions the memory between system and JVM. */
   def apply(cores: Option[Cores] = None, totalMemory: Option[Memory] = None) : SystemResources = {
     val heapSize = Resource.heapSize
-    val system   = totalMemory match {
-      case Some(memory) if heapSize >= memory => throw new IllegalArgumentException(s"Heap size ${heapSize.prettyString} (-Xmx) must less than the total memory of ${memory.prettyString}.")
-      case Some(memory) => memory - heapSize
-      case None         => Memory((Resource.systemMemory.bytes * TaskManagerDefaults.defaultSystemMemoryScalingFactor - heapSize.bytes).toLong)
+
+    val (system, jvm) = totalMemory match {
+      case Some(memory) => (memory, heapSize)
+      case None         => (Resource.systemMemory - heapSize, heapSize)
```

Comment:
Added the check outside the match, in case some nefarious person passes in a 0 or negative value.